src/main/java/net/backupcup/mcde/screen/RunicTableScreen.java

```diff
@@ -61,13 +64,13 @@ protected void init() {
             .withDimPredicate(choice -> {
                 int level = 1;
                 boolean isMaxedOut = false;
-                if (choice instanceof ChoiceWithLevel withLevel) {
+                if (choice instanceof Chosen withLevel) {
```

Comment:
The class of `Chosen` may be better renamed to `Choice` as this is what the class is for. The use of `past participles` as Class names can be very confusing for non-English reviewers as well as being able to tell what the class is for, in the first place. In my mind, `Chosen` sounds like it would be an `interface` rather than a `class`. As such, consider renaming the class to alleviate potential confusion.

Reply:
I see. But didn't you realize that `Choice` already exists? I see why you question this change but I return an instance of this class in `getChosen` method, so it indicates a choice which is already chosen by a player and has level.

I know that this part of code where I check with `instanceof` is not particularly clean. It would require a moderate refactor to rewrite this in a more concise way.